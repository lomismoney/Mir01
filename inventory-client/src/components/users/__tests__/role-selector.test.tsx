import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { RoleSelector } from "../role-selector";

/**
 * RoleSelector 組件測試
 * 
 * 測試範圍：
 * - 基本渲染
 * - 角色選擇功能
 * - 多選功能
 * - 禁用狀態
 * - 錯誤訊息顯示
 * - 回調函數調用
 */
describe("RoleSelector", () => {
  let user: ReturnType<typeof userEvent.setup>;
  const mockOnRolesChange = jest.fn();

  beforeEach(() => {
    user = userEvent.setup();
    jest.clearAllMocks();
  });

  /**
   * 測試基本渲染
   */
  it("應該正確渲染所有角色選項", () => {
    render(
      <RoleSelector
        selectedRoles={[]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 檢查所有角色標籤是否存在
    expect(screen.getByText("管理員")).toBeInTheDocument();
    expect(screen.getByText("員工")).toBeInTheDocument();
    expect(screen.getByText("檢視者")).toBeInTheDocument();
    expect(screen.getByText("安裝師傅")).toBeInTheDocument();
    
    // 檢查描述文字
    expect(screen.getByText("擁有系統的完整管理權限")).toBeInTheDocument();
    expect(screen.getByText("可以管理商品、庫存和訂單")).toBeInTheDocument();
    expect(screen.getByText("只能查看系統資料，不能修改")).toBeInTheDocument();
    expect(screen.getByText("負責安裝工作，可查看和管理安裝單")).toBeInTheDocument();
  });

  /**
   * 測試預選角色的顯示
   */
  it("應該正確顯示已選中的角色", () => {
    render(
      <RoleSelector
        selectedRoles={["admin", "staff"]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 檢查已選中的 checkbox
    const adminCheckbox = screen.getByLabelText(/管理員/);
    const staffCheckbox = screen.getByLabelText(/員工/);
    const viewerCheckbox = screen.getByLabelText(/檢視者/);
    
    expect(adminCheckbox).toBeChecked();
    expect(staffCheckbox).toBeChecked();
    expect(viewerCheckbox).not.toBeChecked();
  });

  /**
   * 測試選擇角色功能
   */
  it("點擊未選中的角色應該添加到選中列表", async () => {
    render(
      <RoleSelector
        selectedRoles={["admin"]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 點擊員工角色
    const staffCheckbox = screen.getByLabelText(/員工/);
    await user.click(staffCheckbox);
    
    // 檢查回調函數是否被正確調用
    expect(mockOnRolesChange).toHaveBeenCalledWith(["admin", "staff"]);
  });

  /**
   * 測試取消選擇角色功能
   */
  it("點擊已選中的角色應該從選中列表移除", async () => {
    render(
      <RoleSelector
        selectedRoles={["admin", "staff", "viewer"]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 點擊已選中的員工角色
    const staffCheckbox = screen.getByLabelText(/員工/);
    await user.click(staffCheckbox);
    
    // 檢查回調函數是否被正確調用（移除 staff）
    expect(mockOnRolesChange).toHaveBeenCalledWith(["admin", "viewer"]);
  });

  /**
   * 測試多選功能
   */
  it("應該支援多選功能", async () => {
    // 測試從已選中一個角色開始
    render(
      <RoleSelector
        selectedRoles={["admin"]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 檢查預選的角色是否正確顯示
    const adminCheckbox = screen.getByLabelText(/管理員/);
    const staffCheckbox = screen.getByLabelText(/員工/);
    
    // 管理員應該已被選中
    expect(adminCheckbox).toBeChecked();
    expect(staffCheckbox).not.toBeChecked();
    
    // 點擊員工角色添加到選中列表
    await user.click(staffCheckbox);
    
    // 應該呼叫 onRolesChange 並包含兩個角色
    expect(mockOnRolesChange).toHaveBeenCalledWith(["admin", "staff"]);
  });

  /**
   * 測試禁用狀態
   */
  it("在禁用狀態下所有 checkbox 都應該被禁用", () => {
    render(
      <RoleSelector
        selectedRoles={["admin"]}
        onRolesChange={mockOnRolesChange}
        disabled={true}
      />
    );
    
    // 檢查所有 checkbox 都被禁用
    const checkboxes = screen.getAllByRole("checkbox");
    checkboxes.forEach(checkbox => {
      expect(checkbox).toBeDisabled();
    });
  });

  /**
   * 測試禁用狀態下點擊無效
   */
  it("在禁用狀態下點擊不應該觸發回調", async () => {
    render(
      <RoleSelector
        selectedRoles={[]}
        onRolesChange={mockOnRolesChange}
        disabled={true}
      />
    );
    
    const adminCheckbox = screen.getByLabelText(/管理員/);
    await user.click(adminCheckbox);
    
    // 禁用狀態下回調不應該被調用
    expect(mockOnRolesChange).not.toHaveBeenCalled();
  });

  /**
   * 測試沒有選擇角色時的錯誤訊息
   */
  it("沒有選擇任何角色時應該顯示錯誤訊息", () => {
    render(
      <RoleSelector
        selectedRoles={[]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 檢查錯誤訊息
    expect(screen.getByText("請至少選擇一個角色")).toBeInTheDocument();
  });

  /**
   * 測試有選擇角色時不顯示錯誤訊息
   */
  it("有選擇角色時不應該顯示錯誤訊息", () => {
    render(
      <RoleSelector
        selectedRoles={["admin"]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 錯誤訊息不應該存在
    expect(screen.queryByText("請至少選擇一個角色")).not.toBeInTheDocument();
  });

  /**
   * 測試角色圖標存在
   */
  it("每個角色都應該有對應的圖標", () => {
    render(
      <RoleSelector
        selectedRoles={[]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 檢查是否有 SVG 圖標（每個角色對應一個圖標）
    const icons = document.querySelectorAll("svg");
    expect(icons.length).toBe(4); // 至少有 4 個角色圖標
  });

  /**
   * 測試 Label 與 Checkbox 的關聯
   */
  it("點擊標籤應該能切換對應的 checkbox", async () => {
    render(
      <RoleSelector
        selectedRoles={[]}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 點擊管理員標籤
    const adminLabel = screen.getByText("管理員");
    await user.click(adminLabel);
    
    // 檢查回調函數是否被調用
    expect(mockOnRolesChange).toHaveBeenCalledWith(["admin"]);
  });

  /**
   * 測試所有角色都被選中的情況
   */
  it("所有角色都被選中時應該正確顯示", () => {
    const allRoles = ["admin", "staff", "viewer", "installer"];
    
    render(
      <RoleSelector
        selectedRoles={allRoles}
        onRolesChange={mockOnRolesChange}
      />
    );
    
    // 檢查所有 checkbox 都被選中
    const checkboxes = screen.getAllByRole("checkbox");
    checkboxes.forEach(checkbox => {
      expect(checkbox).toBeChecked();
    });
    
    // 不應該顯示錯誤訊息
    expect(screen.queryByText("請至少選擇一個角色")).not.toBeInTheDocument();
  });

  /**
   * 測試組件快照
   */
  it("應該匹配快照 - 無選擇狀態", () => {
    const { container } = render(
      <RoleSelector
        selectedRoles={[]}
        onRolesChange={mockOnRolesChange}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  /**
   * 測試組件快照 - 有選擇狀態
   */
  it("應該匹配快照 - 有選擇狀態", () => {
    const { container } = render(
      <RoleSelector
        selectedRoles={["admin", "staff"]}
        onRolesChange={mockOnRolesChange}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  /**
   * 測試組件快照 - 禁用狀態
   */
  it("應該匹配快照 - 禁用狀態", () => {
    const { container } = render(
      <RoleSelector
        selectedRoles={["admin"]}
        onRolesChange={mockOnRolesChange}
        disabled={true}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
}); 
