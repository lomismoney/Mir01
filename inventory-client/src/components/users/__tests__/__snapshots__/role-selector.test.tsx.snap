// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleSelector 應該匹配快照 - 有選擇狀態 1`] = `
<div
  class="space-y-3"
>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="true"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="checked"
      id="role-admin"
      role="checkbox"
      type="button"
      value="on"
    >
      <span
        class="flex items-center justify-center text-current"
        data-state="checked"
        style="pointer-events: none;"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          />
        </svg>
      </span>
    </button>
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-admin"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shield h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 13c0 5-3.5 7.5-7.66 8.95a1 1 0 0 1-.67-.01C7.5 20.5 4 18 4 13V6a1 1 0 0 1 1-1c2 0 4.5-1.2 6.24-2.72a1.17 1.17 0 0 1 1.52 0C14.51 3.81 17 5 19 5a1 1 0 0 1 1 1z"
          />
        </svg>
        管理員
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        擁有系統的完整管理權限
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="true"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="checked"
      id="role-staff"
      role="checkbox"
      type="button"
      value="on"
    >
      <span
        class="flex items-center justify-center text-current"
        data-state="checked"
        style="pointer-events: none;"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          />
        </svg>
      </span>
    </button>
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-staff"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-store h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2 7 4.41-4.41A2 2 0 0 1 7.83 2h8.34a2 2 0 0 1 1.42.59L22 7"
          />
          <path
            d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
          />
          <path
            d="M15 22v-4a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2v4"
          />
          <path
            d="M2 7h20"
          />
          <path
            d="M22 7v3a2 2 0 0 1-2 2a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 16 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 12 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 8 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 4 12a2 2 0 0 1-2-2V7"
          />
        </svg>
        員工
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        可以管理商品、庫存和訂單
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="unchecked"
      id="role-viewer"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-viewer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-eye h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
        檢視者
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        只能查看系統資料，不能修改
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="unchecked"
      id="role-installer"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-installer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-square-pen h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
          />
        </svg>
        安裝師傅
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        負責安裝工作，可查看和管理安裝單
      </p>
    </div>
  </div>
</div>
`;

exports[`RoleSelector 應該匹配快照 - 無選擇狀態 1`] = `
<div
  class="space-y-3"
>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="unchecked"
      id="role-admin"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-admin"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shield h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 13c0 5-3.5 7.5-7.66 8.95a1 1 0 0 1-.67-.01C7.5 20.5 4 18 4 13V6a1 1 0 0 1 1-1c2 0 4.5-1.2 6.24-2.72a1.17 1.17 0 0 1 1.52 0C14.51 3.81 17 5 19 5a1 1 0 0 1 1 1z"
          />
        </svg>
        管理員
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        擁有系統的完整管理權限
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="unchecked"
      id="role-staff"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-staff"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-store h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2 7 4.41-4.41A2 2 0 0 1 7.83 2h8.34a2 2 0 0 1 1.42.59L22 7"
          />
          <path
            d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
          />
          <path
            d="M15 22v-4a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2v4"
          />
          <path
            d="M2 7h20"
          />
          <path
            d="M22 7v3a2 2 0 0 1-2 2a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 16 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 12 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 8 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 4 12a2 2 0 0 1-2-2V7"
          />
        </svg>
        員工
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        可以管理商品、庫存和訂單
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="unchecked"
      id="role-viewer"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-viewer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-eye h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
        檢視者
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        只能查看系統資料，不能修改
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-state="unchecked"
      id="role-installer"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-installer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-square-pen h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
          />
        </svg>
        安裝師傅
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        負責安裝工作，可查看和管理安裝單
      </p>
    </div>
  </div>
  <p
    class="text-sm text-destructive"
  >
    請至少選擇一個角色
  </p>
</div>
`;

exports[`RoleSelector 應該匹配快照 - 禁用狀態 1`] = `
<div
  class="space-y-3"
>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="true"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-disabled=""
      data-state="checked"
      disabled=""
      id="role-admin"
      role="checkbox"
      type="button"
      value="on"
    >
      <span
        class="flex items-center justify-center text-current"
        data-disabled=""
        data-state="checked"
        style="pointer-events: none;"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-check h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 6 9 17l-5-5"
          />
        </svg>
      </span>
    </button>
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-admin"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shield h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 13c0 5-3.5 7.5-7.66 8.95a1 1 0 0 1-.67-.01C7.5 20.5 4 18 4 13V6a1 1 0 0 1 1-1c2 0 4.5-1.2 6.24-2.72a1.17 1.17 0 0 1 1.52 0C14.51 3.81 17 5 19 5a1 1 0 0 1 1 1z"
          />
        </svg>
        管理員
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        擁有系統的完整管理權限
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-disabled=""
      data-state="unchecked"
      disabled=""
      id="role-staff"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-staff"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-store h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2 7 4.41-4.41A2 2 0 0 1 7.83 2h8.34a2 2 0 0 1 1.42.59L22 7"
          />
          <path
            d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
          />
          <path
            d="M15 22v-4a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2v4"
          />
          <path
            d="M2 7h20"
          />
          <path
            d="M22 7v3a2 2 0 0 1-2 2a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 16 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 12 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 8 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 4 12a2 2 0 0 1-2-2V7"
          />
        </svg>
        員工
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        可以管理商品、庫存和訂單
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-disabled=""
      data-state="unchecked"
      disabled=""
      id="role-viewer"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-viewer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-eye h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
        檢視者
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        只能查看系統資料，不能修改
      </p>
    </div>
  </div>
  <div
    class="flex items-start space-x-3"
  >
    <button
      aria-checked="false"
      class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground mt-1"
      data-disabled=""
      data-state="unchecked"
      disabled=""
      id="role-installer"
      role="checkbox"
      type="button"
      value="on"
    />
    <div
      class="grid gap-1.5 leading-none"
    >
      <label
        class="text-sm leading-none select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 flex items-center gap-2 font-medium cursor-pointer"
        data-slot="label"
        for="role-installer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-square-pen h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
          />
        </svg>
        安裝師傅
      </label>
      <p
        class="text-sm text-muted-foreground"
      >
        負責安裝工作，可查看和管理安裝單
      </p>
    </div>
  </div>
</div>
`;
